shuup/admin/modules/orders/views/shipment.py

```diff
@@ -97,6 +97,15 @@ def form_invalid(self, form):
         add_form_errors_as_messages(self.request, form)
         return super(OrderCreateShipmentView, self).form_invalid(form)
 
+    def create_shipment(self, order, product_quantities, shipment):
+        """
+        This function exists so subclasses can implement custom logic without overriding the entire form_valid method
```

Comment:
You could actually write open the parameters. For example here it seems to be little open what the `product_quantities`.